content/cloud-docs/registry/publish-providers.mdx

```diff
@@ -0,0 +1,115 @@
+
+Private providers do not currently support documentation.
+
+### Signed Releases
+
+GPG signing is required for custom providers, and you must upload the public key of the GPG keypair used to sign the release. Refer to [Preparing and Adding a Signing Key](/registry/providers/publishing#preparing-and-adding-a-signing-key) for more details. Unlike the public Terraform Registry, the private registry does not automatically upload new releases. You must manually add new provider versions and the associated release files.
+
+## Publishing a Provider and Creating a Version
+
+You must create a provider, at least one version, and at least one platform for that version before consumers can use your provider in configurations. Before you begin, confirm that you can access the provider binaries, public GPG signing key, `SHA256SUMS` file, and `SHA256SUMS.sig` file from at least one release. You will upload these artifacts in the following steps.
+
+To publish a new provider through the Terraform Cloud API:
+
+1. Use the [Upload a GPG key endpoint](/cloud-docs/api-docs/providers#upload-a-gpg-key-private-provider) to add the public key that matches the signing key for the release. The response contains a `key-id` that you will use to create a provider version.
+
+    ```json
+    "key-id": "32966F3FB5AC1129"
+    ```
+1. Use the [Create a Provider Version endpoint](/cloud-docs/api-docs/providers#create-a-provider-version) to create a version with the `key-id` from the previous step. The response includes URL links that you will use to upload the `SHA256SUMS` and `SHA256.sig` files.
+
+    ```json
+    "links": {
+        "shasums-upload":     "https://archivist.terraform.io/v1/object/dmF1b64hd73ghd63",
+        "shasums-sig-upload": "https://archivist.terraform.io/v1/object/dmF1b37dj37dh33d"
+      }
+    ```
+
+1. Upload the `SHA256SUMS` and `SHA256SUMS.sig` files to the URLs returned in the previous step. The example command below uploads the files from a local machine.
+
+    ```
+    $ curl -T local-example/terraform-provider-random_3.1.3_SHA256SUMS
+      https://archivist.terraform.io/v1/object/dmF1b64hd73ghd63
+    ```
+
+1. Use the [Upload a Provider Platform](/cloud-docs/api-docs/providers#upload-a-provider-platform-private-provider) endpoint to create a platform for the version. Platforms are binaries that allow the provider to run on a particular operating system and architecture combination (e.g., Linux and AMD64). The response includes a `provider-binary-upload` URL that you will use to upload the binary file for the platform.
```

Comment:
```suggestion
1. Use the [Add a Provider Platform](/cloud-docs/api-docs/providers#add-a-provider-platform-private-provider) endpoint to create a platform for the version. Platforms are binaries that allow the provider to run on a particular operating system and architecture combination (e.g., Linux and AMD64). The response includes a `provider-binary-upload` URL that you will use to upload the binary file for the platform.
```

Reply:
i got you! I'll make sure this is updated :)

Reply:
🙌